raiden/network/transport/matrix/transport.py

```diff
@@ -982,11 +982,10 @@ def _handle_invite(self, room_id: RoomID, state: dict) -> None:
         room.invite_only = private_room
         self._set_room_id_for_address(address=peer_address, room_id=room_id)
 
-    def _handle_member_join(self, room_id: RoomID) -> None:
-        room = self._client.rooms[room_id]
+    def _handle_member_join(self, room: Room) -> None:
         partner_addresses = self._extract_partner_addresses(room.get_joined_members())
 
-        if len(partner_addresses) > 1:
+        if len(partner_addresses) > 1 and not self._is_broadcast_room(room):
```

Comment:
Suggestion: Move the is broadcast room check to the beginning of the function. That way we save extracting the addresses for all peers (which can be a lot in broadcast rooms).